### PR TITLE
added acknowledgements + fixed encoder training

### PR DESCRIPTION
**Pull request by soumik12345**

### CODE REVIEW

**Comment by soumik12345** [comment]
https://github.com/soumik12345/tf2_gans/issues/14

File: gaugan/models/gaugan.py [hunk 1]
```diff
@@ -153,7 +154,7 @@ def train_generator(
 
         gradients = tape.gradient(total_loss, self.combined_model.trainable_variables)
```
Comment (yonigottesman) [inline gaugan/models/gaugan.py:155]: add + self.encoder.trainable_variables to tape.gradient

File: gaugan/models/gaugan.py [hunk 2]
```diff
@@ -1,6 +1,7 @@
 """References:
 - https://github.com/PacktPublishing/Hands-On-Image-Generation-with-TensorFlow-2.0/blob/master/Chapter06/ch6_gaugan.ipynb
 - https://www.packtpub.com/product/hands-on-image-generation-with-tensorflow/9781838826789
```
Comment (sayakpaul) [inline gaugan/models/gaugan.py:3]: Let's have some text added and then hyperlink them with these links.

**Review by sayakpaul — CHANGES_REQUESTED** [review]

File: gaugan/models/gaugan.py (hunk 1 above)
Comment (sayakpaul) [reply gaugan/models/gaugan.py:155]: https://github.com/soumik12345/tf2_gans/issues/14#issuecomment-1034590292

File: gaugan/models/gaugan.py [hunk 3]
```diff
@@ -153,7 +154,7 @@ def train_generator(
 
         gradients = tape.gradient(total_loss, self.combined_model.trainable_variables)
         self.generator_optimizer.apply_gradients(
-            zip(gradients, self.combined_model.trainable_variables)
+            zip(gradients, self.combined_model.trainable_variables + self.encoder.trainable_variables)
```
Comment (sayakpaul) [inline gaugan/models/gaugan.py:157]: https://github.com/soumik12345/tf2_gans/issues/14#issuecomment-1034590292

**Review by sayakpaul — CHANGES_REQUESTED** [review]

**Comment by soumik12345** [comment]
@sayakpaul Everything looks good to me.

**Comment by yonigottesman** [comment]
thanks, the tutorial is great 🙏
will you also update the keras.io version?

**Comment by sayakpaul** [comment]
We have a PR in progress to update these recent changes.
